jekyll/_cci2/docker-hub-pull-through-mirror.adoc

```diff
@@ -0,0 +1,73 @@
+---
+version:
+- Server v2.x
+- Server Admin
+---
+= Setting Up Docker Hub Pull Through Mirror
+:page-layout: classic-docs
+:page-liquid:
+:icons: font
+:toc: macro
+:toc-title:
+
+This section describes how to configure a Docker Hub pull through registry mirror.
+
+toc::[]
+
+== Overview
+
+Starting [November 1, 2020](https://www.docker.com/blog/scaling-docker-to-serve-millions-more-developers-network-egress/), Docker Hub will impose rate limits for anonymous pulls based on the originating IP. To avoid service disruption, it is recommended to authenticate Docker pulls with Docker Hub. You can authenticate using build config see [Using Docker Authenticated Pulls](https://circleci.com/docs/2.0/private-images/).
+
+Alternatively, you can set up a Docker Hub pull through registry mirror pre-configured with Docker Hub account credentials for your entire Server install.
+
+Using a pull through registry mirror may also bring additional performance improvements since network roundtrips to Docker Hub are reduced.
+
+These instructions are divided into two parts: Setting up a pull through registry mirror, and configureing Nomad clients to use the mirror.
+
+== Setup a pull through cache registry (proxy for Docker Hub)
+
+[arabic]
+. *Setup a Docker Hub account* which will be used by the pull through cache registry to access Docker Hub.
++
+Tips: Setting up a paid account can be an option to ease the rate limiting further. See also: https://www.docker.com/pricing
+. *Setup an independent Linux server* that has Docker installed.
++
+Hereby  we setup  this  registry as  an independent  server  (i.e. outside  of CircleCI boxes)  to avoid load on  this cache server affecting  other CircleCI Server services.
++
+Assume that the IP address for this server is `192.0.2.1`.
+. Run the command below to *start the cache server*.
++
+*Note*: Replace `DOCKER_HUB_USERNAME` and `DOCKER_HUB_ACCESS_TOKEN` with the username for the Docker Hub account and the access token you obtained through https://hub.docker.com/settings/security, respectively.
++
+`sudo docker run -d -p 80:5000 --restart=always --name=through-cache -e REGISTRY_PROXY_REMOTEURL="https://registry-1.docker.io" -e REGISTRY_PROXY_USERNAME=DOCKER_HUB_USERNAME -e REGISTRY_PROXY_PASSWORD=DOCKER_HUB_ACCESS_TOKEN registry`
+
```

Comment:
@lewang I would like to add another note here that `80/tcp` should be open (apologies this is a missing piece in my original version 🙇 ). It should look like this:

```
. Finally, make sure that the TCP port 80 (i.e., HTTP) is open and accessible. For better security, we recommend to open the port only to Nomad clients.
** On AWS, for example, you will need to make sure that both Security Groups and the firewall at the OS level are configured correctly to accept connections from Nomad clients over HTTP.
```